src/core/border_router/routing_manager.cpp

```diff
@@ -3767,6 +3791,41 @@ Error RoutingManager::PdPrefixManager::Process(const RouterAdvert::RxMessage &aM
     return error;
 }
 
+bool RoutingManager::PdPrefixManager::ProcessPrefixEntry(DiscoveredPrefixTable::Entry &aEntry,
+                                                         DiscoveredPrefixTable::Entry &aFavoredEntry)
+{
+    bool currentPrefixUpdated = false;
+
+    if (!IsValidPdPrefix(aEntry.GetPrefix()))
+    {
+        LogWarn("PdPrefixManager: Ignore invalid PIO entry %s", aEntry.GetPrefix().ToString().AsCString());
```

Comment:
Since it may not be from PIO?
```suggestion
        LogWarn("PdPrefixManager: Ignore invalid prefix %s", aEntry.GetPrefix().ToString().AsCString());
```